webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/editablemapping/EditableMappingUpdater.scala

```diff
@@ -349,19 +351,31 @@ class EditableMappingUpdater(
                          agglomerateGraph2: AgglomerateGraph,
                          segmentId1: Long,
                          segmentId2: Long): Option[AgglomerateGraph] = {
-    val segment1IsValid = agglomerateGraph1.segments.contains(segmentId1)
-    val segment2IsValid = agglomerateGraph2.segments.contains(segmentId2)
-    if (segment1IsValid && segment2IsValid) {
+    val newEdgeAffinity = 255.0f
+    val newEdge = AgglomerateEdge(segmentId1, segmentId2)
+    if (agglomerateGraph1 == agglomerateGraph2) {
+      // Agglomerate is merged with itself. Insert new edge anyway, if it does not exist yet
       val newEdge = AgglomerateEdge(segmentId1, segmentId2)
-      val newEdgeAffinity = 255.0f
-      Some(
-        AgglomerateGraph(
-          segments = agglomerateGraph1.segments ++ agglomerateGraph2.segments,
-          edges = newEdge +: (agglomerateGraph1.edges ++ agglomerateGraph2.edges),
-          affinities = newEdgeAffinity +: (agglomerateGraph1.affinities ++ agglomerateGraph2.affinities),
-          positions = agglomerateGraph1.positions ++ agglomerateGraph2.positions
-        ))
-    } else None
+      if (agglomerateGraph1.edges.contains(newEdge)) {
+        Some(agglomerateGraph1)
+      } else {
+        Some(
+          agglomerateGraph1.copy(edges = newEdge +: agglomerateGraph1.edges,
+                                 affinities = newEdgeAffinity +: agglomerateGraph1.affinities))
+      }
+    } else {
+      val segment1IsValid = agglomerateGraph1.segments.contains(segmentId1)
+      val segment2IsValid = agglomerateGraph2.segments.contains(segmentId2)
+      if (segment1IsValid && segment2IsValid) {
```

Comment:
Could you please add a comment about what kind of potential buggy behaviour this if catches / avoids?
The if above handles repeated merge operations. 
So this catches in case e.g. `segmentId1` was split off in another annotation that is merged into this annotation?

Reply:
I can’t properly explain it. In fact I think it may be obsolete. Wrote follow-up issue https://github.com/scalableminds/webknossos/issues/8368 to track double-checking that